homeassistant/components/hyundai_kia_connect/binary_sensor.py

```diff
@@ -0,0 +1,206 @@
+"""Sensor for Hyundai / Kia Connect integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Final
+
+from hyundai_kia_connect_api import Vehicle
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+
+from .const import DOMAIN
+from .entity import HyundaiKiaConnectEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class HyundaiKiaBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A class that describes custom binary sensor entities."""
+
+    on_icon: str | None = None
+    off_icon: str | None = None
+
+
```

Comment:
```suggestion
```

All cases setting icons use device classes, thus should not set icons at all. Making this obsolete.